.github/workflows/macos.yml

```
@@ -71,25 +71,31 @@ jobs:
 
       - run: make
 
+      - name: Set test options for skipped tests
+        run: |
+          set -ex
+          TESTS="$(echo "${{ matrix.skipped_tests }}" | sed 's| |$$/ -n!/|g;s|^|-n!/|;s|$|$$/|')"
+          echo "TESTS=${TESTS}" >> $GITHUB_ENV
+        if: ${{ matrix.test_task == 'check' && matrix.skipped_tests }}
+
       - name: make ${{ matrix.test_task }}
         run: |
-          make -s ${{ matrix.test_task }} ${TESTS:+TESTS=`echo "$TESTS" | sed 's| |$$/ -n!/|g;s|^|-n!/|;s|$|$$/|'`}
+          make -s ${{ matrix.test_task }} ${TESTS:+TESTS="$(echo "$TESTS")"}
```

This command substitution seems redundant.

```suggestion
          make -s ${{ matrix.test_task }} ${TESTS:+"$TESTS"}
```

---

Ah you are right! I will fix it.

---

@nobu I assume you intended the following change. I suppose we still need to the `TESTS=` part.

```diff
diff --git a/.github/workflows/macos.yml b/.github/workflows/macos.yml
index 2ac5188bb7..73961b9fae 100644
--- a/.github/workflows/macos.yml
+++ b/.github/workflows/macos.yml
@@ -89,7 +89,7 @@ jobs:
 
       - name: make ${{ matrix.test_task }}
         run: |
-          make -s ${{ matrix.test_task }} ${TESTS:+TESTS="$(echo "$TESTS")"}
+          make -s ${{ matrix.test_task }} ${TESTS:+TESTS="$TESTS"}
         timeout-minutes: 60
         env:
           RUBY_TESTOPTS: '-q --tty=no'
```

---

I fixed it.